### PR TITLE
Add merge functionality to config_helpers script

### PR DESCRIPTION
**Pull request by JacobCallahan**
This change add the ability for the script to merge in changes from another config directiory. This will most likely be used to merge in changes from the upstream config repository.

I also made some optimisations to the existing symlink functionality.

### CODE REVIEW

**Review by Griffin-Sullivan — COMMENTED** [review]
Tested merge functionality and it worked successfully. I'm not exactly sure if the symlink is working for me though.

**Comment by Griffin-Sullivan** [comment]
When I look for the symlinks they aren't showing up.

**Review by Griffin-Sullivan — APPROVED** [review]
Nevermind I was misunderstanding the purpose of the symlink. Everything is working correctly. I deleted a file and reran the symlink and it worked.

**Comment by JacobCallahan** [comment]
@Griffin-Sullivan thanks for checking!

File: scripts/config_helpers.py [hunk 1]
```diff
@@ -2,16 +2,66 @@
...
+                original[key] = merge_nested_dictionaries(original[key], value, overwrite)
+        elif original[key] == value:
+            continue
+        # if the key is in the original, and the value is a list, ask the user
+        elif overwrite == "ask":
+            choice_prompt = (
+                f"The current value for {key} is {original[key]}.\n"
+                "Please choose an option:\n"
+                "1. Keep the current value\n"
+                f"2. Overwrite the current value with {value}\n"
```
Comment (jyejare) [inline scripts/config_helpers.py:30]:
```suggestion
                f"2. Overwrite the current value with new/upstream value {value}\n"
```

File: scripts/config_helpers.py [hunk 2]
```diff
@@ -21,19 +71,56 @@ def config():
...
+@click.option(
+    "--from",
+    "from_",
+    type=click.Path(exists=True, file_okay=False),
+    help="The path to the existing config directory",
+)
+@click.option(
+    "--strategy",
+    type=click.Choice(["ask", "overwrite", "preserve"]),
+    help="The strategy to use when merging",
```
Comment (jyejare) [inline scripts/config_helpers.py:93]:
```suggestion
    help="The strategy to use when merging. Choose one from ask, overwrite, or preserve. If not provided then default is ask",
```

**Review by jyejare — APPROVED** [review]
Looks Great! 

Left some nitpick comments !

**Review by jyejare — COMMENTED** [review]
@JacobCallahan  Also why dont we take this config helper to `Makefile` so that we let robottelo users that something like this is available and `make <tab><tab>` this would be shown in their terminal ?

**Comment by jyejare** [comment]
@JacobCallahan  TBH I dont use symlinks between CI and Robottelo files because any changed I make to those file from robottelo appears in CI confs and whenever I want to open a new PR in CI (satelliteqe-jenkins) , the git status just pops these files and it make my life difficult to raise a PR. How do we handle this with symlinks here then ? Or should we altogether remove the symlinks and copy plain and merge when there is a change in CI/upstream ?

**Comment by JacobCallahan** [comment]
@jyejare this change allows you to do either behavior. A user, who doesn't make changes to config, can symlink files they don't touch. Also, they can use this script to copy/merge changes from the ci repo to their local robottelo.

**Review by jyejare — APPROVED** [review]

**Comment by jyejare** [comment]
trigger: test-robottelo

**Comment by jyejare** [comment]
@JacobCallahan CI checks looks to be failing. Please look into it so we can merge !

File: scripts/config_helpers.py (hunk 1 above)
Comment (JacobCallahan) [reply scripts/config_helpers.py:30]: I think new/upstream is implied

File: scripts/config_helpers.py (hunk 2 above)
Comment (JacobCallahan) [reply scripts/config_helpers.py:93]: I can't check right now, but are the options not presented by click in the help text?

**Comment by jyejare** [comment]
trigger: test-robottelo

**Pull request merged**